tests/common/ixia/port.py

```diff
@@ -0,0 +1,89 @@
+import enum
+
+class IxiaPortType(enum.Enum):
+    """
+    IXIA port type
+    """
+    IPInterface = 1
+    PortChannelMember = 2
+    VlanMember = 3
+
+class IxiaPortConfig:
+    """
+    IXIA port configuration information
+    """
+    def __init__(self, id, ip, mac, gw, gw_mac, prefix_len, port_type, peer_port):
+        self.id = id
+        self.ip = ip
+        self.mac = mac
+        self.gateway = gw
+        self.gateway_mac = gw_mac
+        self.prefix_len = prefix_len
+        self.type = port_type
+        self.peer_port = peer_port
+
+def select_ports(port_config_list, duthost, pattern, rx_port_id):
+    """
+    Given a traffic pattern, select IXIA ports to send and receive traffic
+
+    Args:
+        port_config_list (list of IxiaPortConfig):
+        duthost (AnsibleHost): Device Under Test (DUT)
+        pattern (str): traffic pattern, "many to one" or "all to all"
+        rx_port_id (int): ID of the port that should receive traffic, e.g.,
+        recever in "many to one" traffic pattern
+
+    Returns:
+        tx_port_id_list (list): IDs of IXIA ports to send traffic
+        rx_port_id_list (list): IDs of IXIA ports to receive traffic
+    """
+    tx_port_id_list = []
+    rx_port_id_list = []
+
+    patterns = ['all to all', 'many to one']
+    if pattern not in patterns:
+        raise ValueError('invalid traffic pattern passed in "{}", must be {}'.format(
+            pattern, ' or '.join(['"{}"'.format(src) for src in patterns])))
+
+    rx_port_config = next((x for x in port_config_list if x.id == rx_port_id), None)
+    if rx_port_config is None:
+        raise ValueError('Fail to find configuration for RX port')
+
+    if pattern == "many to one":
+        rx_port_id_list = [rx_port_id]
+        """ Interfaces in the same portchannel cannot send traffic to each other """
+        if rx_port_config.type == IxiaPortType.PortChannelMember:
+            tx_port_id_list = [x.id for x in port_config_list \
+                               if x.ip != rx_port_config.ip]
+        else:
+            tx_port_id_list = [x.id for x in port_config_list if x.id != rx_port_id]
+
+    active_port_id_list = tx_port_id_list + rx_port_id_list
+    for port_config in port_config_list:
+        dut_port = port_config.peer_port
+        if port_config.id in active_port_id_list:
+            duthost.shell('sudo config interface startup {}'.format(dut_port))
+        else:
+            duthost.shell('sudo config interface shutdown {}'.format(dut_port))
```

Comment:
We don't need to do this anymore since we are only using single portchannel member